WARNING:absl:No GPU/TPU found, falling back to CPU. (Set TF_CPP_MIN_LOG_LEVEL=0 and rerun for more info.)
-0.4
Computing evidence...
Avg evidence: -11.20989033006857
Traceback (most recent call last):
  File "train_nonlinear.py", line 182, in <module>
    main(args, save_dir)
  File "train_nonlinear.py", line 100, in main
    trainer = SVITrainer(p=p, 
  File "/home/infres/chagneux/repos/backward_ica/backward_ica/svi.py", line 321, in __init__
    self.q.print_num_params()
  File "/home/infres/chagneux/repos/backward_ica/backward_ica/hmm.py", line 752, in print_num_params
    params = self.get_random_params(random.PRNGKey(0))
  File "/home/infres/chagneux/repos/backward_ica/backward_ica/hmm.py", line 730, in get_random_params
    filt_update_params = self.filt_update_init_params(subkeys[2], dummy_obs, dummy_pred_state)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/haiku/_src/transform.py", line 113, in init_fn
    params, state = f.init(*args, **kwargs)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/haiku/_src/transform.py", line 381, in init_fn
    f(*args, **kwargs)
  File "/home/infres/chagneux/repos/backward_ica/backward_ica/hmm.py", line 688, in johnson_update_forward
    return GaussianParams(eta1 = eta1 + pred_state.eta1, eta2 = eta2 + pred_state.eta2)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/_src/numpy/lax_numpy.py", line 4538, in deferring_binary_op
    return binary_op(self, other)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/_src/traceback_util.py", line 162, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/_src/api.py", line 466, in cache_miss
    out_flat = xla.xla_call(
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/core.py", line 1771, in bind
    return call_bind(self, fun, *args, **params)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/core.py", line 1787, in call_bind
    outs = top_trace.process_call(primitive, fun_, tracers, params)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/core.py", line 660, in process_call
    return primitive.impl(f, *tracers, **params)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/_src/dispatch.py", line 149, in _xla_call_impl
    compiled_fun = _xla_callable(fun, device, backend, name, donated_invars,
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/linear_util.py", line 285, in memoized_fun
    ans = call(fun, *args)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/_src/dispatch.py", line 197, in _xla_callable_uncached
    return lower_xla_callable(fun, device, backend, name, donated_invars,
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/_src/profiler.py", line 206, in wrapper
    return func(*args, **kwargs)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/_src/dispatch.py", line 228, in lower_xla_callable
    jaxpr, out_avals, consts = pe.trace_to_jaxpr_final(
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/_src/profiler.py", line 206, in wrapper
    return func(*args, **kwargs)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/interpreters/partial_eval.py", line 1844, in trace_to_jaxpr_final
    jaxpr, out_avals, consts = trace_to_subjaxpr_dynamic(
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/interpreters/partial_eval.py", line 1816, in trace_to_subjaxpr_dynamic
    ans = fun.call_wrapped(*in_tracers_)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/linear_util.py", line 168, in call_wrapped
    ans = self.f(*args, **dict(self.params, **kwargs))
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/_src/numpy/ufuncs.py", line 87, in fn
    return lax_fn(x1, x2) if x1.dtype != np.bool_ else bool_lax_fn(x1, x2)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/_src/lax/lax.py", line 402, in add
    return add_p.bind(x, y)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/core.py", line 323, in bind
    return self.bind_with_trace(find_top_trace(args), args, params)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/core.py", line 326, in bind_with_trace
    out = trace.process_primitive(self, map(trace.full_raise, args), params)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/interpreters/partial_eval.py", line 1511, in process_primitive
    return self.default_process_primitive(primitive, tracers, params)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/interpreters/partial_eval.py", line 1515, in default_process_primitive
    out_avals, effects = primitive.abstract_eval(*avals, **params)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/core.py", line 359, in abstract_eval_
    return abstract_eval(*args, **kwargs), no_effects
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/_src/lax/utils.py", line 66, in standard_abstract_eval
    return core.ShapedArray(shape_rule(*avals, **kwargs),
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/_src/lax/lax.py", line 1442, in _broadcasting_shape_rule
    raise TypeError(f'{name} got incompatible shapes for broadcasting: '
jax._src.traceback_util.UnfilteredStackTrace: TypeError: add got incompatible shapes for broadcasting: (10,), (5,).

The stack trace below excludes JAX-internal frames.
The preceding is the original exception that occurred, unmodified.

--------------------

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "train_nonlinear.py", line 182, in <module>
    main(args, save_dir)
  File "train_nonlinear.py", line 100, in main
    trainer = SVITrainer(p=p, 
  File "/home/infres/chagneux/repos/backward_ica/backward_ica/svi.py", line 321, in __init__
    self.q.print_num_params()
  File "/home/infres/chagneux/repos/backward_ica/backward_ica/hmm.py", line 752, in print_num_params
    params = self.get_random_params(random.PRNGKey(0))
  File "/home/infres/chagneux/repos/backward_ica/backward_ica/hmm.py", line 730, in get_random_params
    filt_update_params = self.filt_update_init_params(subkeys[2], dummy_obs, dummy_pred_state)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/haiku/_src/transform.py", line 113, in init_fn
    params, state = f.init(*args, **kwargs)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/haiku/_src/transform.py", line 381, in init_fn
    f(*args, **kwargs)
  File "/home/infres/chagneux/repos/backward_ica/backward_ica/hmm.py", line 688, in johnson_update_forward
    return GaussianParams(eta1 = eta1 + pred_state.eta1, eta2 = eta2 + pred_state.eta2)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/_src/numpy/lax_numpy.py", line 4538, in deferring_binary_op
    return binary_op(self, other)
  File "/home/infres/chagneux/envs/backward_ica/lib/python3.8/site-packages/jax/_src/numpy/ufuncs.py", line 87, in fn
    return lax_fn(x1, x2) if x1.dtype != np.bool_ else bool_lax_fn(x1, x2)
TypeError: add got incompatible shapes for broadcasting: (10,), (5,).
